matching-funds/app/daffy.server.ts

```diff
@@ -0,0 +1,88 @@
+import { z } from "zod";
+
+let BASE_URL = new URL("http://localhost:3000/");
+
+export class DaffyClient {
+  private token: string;
+  private type: "oauth" | "api-key";
+
+  constructor(token: string) {
+    this.token = token;
+    if (token.startsWith("daffy_production_")) {
+      this.type = "api-key";
+    } else {
+      this.type = "oauth";
+    }
+  }
+
+  private get headers() {
+    let headers = new Headers({
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    });
+
+    if (this.type === "oauth") headers.set("Authorization", this.token);
+    else headers.set("X-Api-Key", this.token);
```

Comment:
plz use curly brackets

```suggestion
    if (this.type === "oauth") {
      headers.set("Authorization", this.token);
    } else {
      headers.set("X-Api-Key", this.token);
    }
```

or better yet...

```suggestion
    const headerName = this.type === "oauth" ? "Authorization" : "X-Api-Key";
    headers.set(headerName, this.token);
```

I sure wish we didn't have to do this at all 😞

Reply:
I'm already working on the API client that will do this for you anyway, so you can do `new DaffyKit(token)` and that `token` can be an API key or OAuth token.